### PR TITLE
[SPARK-25944][R][BUILD] AppVeyor change to latest R version (3.5.1)

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

R 3.5.1 is released 2018-07-02. This PR targets to changes R version from 3.4.1 to 3.5.1.

## How was this patch tested?

AppVeyor

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.
I also check the AppVeyor logs.
```
R version 3.5.1 (2018-07-02)
```

**Review by srowen — APPROVED** [review]

File: dev/appveyor-install-dependencies.ps1 [hunk 1]
```diff
@@ -115,7 +115,7 @@ $env:Path += ";$env:HADOOP_HOME\bin"
 Pop-Location
 
 # ========================== R
-$rVer = "3.4.1"
+$rVer = "3.5.1"
 $rToolsVer = "3.4.0"
```
Comment (felixcheung) [inline dev/appveyor-install-dependencies.ps1]: is there update to rtool?

**Review by felixcheung — COMMENTED** [review]
cool!

File: dev/appveyor-install-dependencies.ps1 (hunk 1 above)
Comment (HyukjinKwon) [reply dev/appveyor-install-dependencies.ps1]: Oh~ latest looks 3.5. Let me try to update aswell shortly.

**Pull request closed without merging**